enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/ImageEnricher.java

```diff
@@ -198,4 +266,22 @@ private void mergeImagePullPolicy(ImageConfiguration imageConfiguration, Contain
         }
     }
 
+    private void mergeEnvVariables(Container container) {
+        List<EnvVar> env = container.getEnv();
+        if (env == null) {
+            env = new LinkedList<>();
+            container.setEnv(env);
+        }
+
+        ResourceConfig resource = getContext().getResources();
+        Map<String, String> userEnv = resource != null ? resource.getEnv() : null;
+        if (userEnv != null) {
+            for(Map.Entry<String, String> entry : userEnv.entrySet()) {
+                boolean set = KubernetesResourceUtil.setEnvVar(env, entry.getKey(), entry.getValue(), false);
```

Comment:
Just a minor thing (not required, though): If we could change setEnvVar to return the old value here (and then check getValue() vs. oldValue), we could add both values in the warning messages so that it becomes easier to detect the origin of the values in order to fix the warning.